text/0000-scaling-use-cases.md

```diff
@@ -0,0 +1,115 @@
+- Feature Name: scaling_use_cases
+- Start Date: 2021-07-21
+- RFC PR: (leave this empty)
+- Hathor Issue: (leave this empty)
+- Author: Marcelo Salhab Brogliato <msbrogli@hathor.network>
+
+# Summary
+[summary]: #summary
+
+This document discusses different approches to integrate applications with Hathor where the number of users might grow indefinitely.
+
+# Motivation
+[motivation]: #motivation
+
+There are multiple ways to developing such integrations and it is not trivial to decide which approach works best for each case.
+
+# Guide-level explanation
+[guide-level-explanation]: #guide-level-explanation
+
+The context here is that a company has a growing user base, and each user has one or more addresses in Hathor. We can say that each user has a wallet, but we need to carefully understand what "wallet" means in this case.
+
+The company has custody of its users' tokens. In other words, the users' wallets are fully controlled by the company.
+
+Let's explore different ways of generating and maintaining the users' wallets.
+
+<!-- ##### -->
+<!-- ##### -->
+
+## Global Wallet
+
+In this case, the company will run one global wallet. Each user will be associated to one or more addresses of the global wallet through their indexes.
+
+The global wallet has an unlimited number of enumerable addresses. Each index will be associated to a user. For instance:
+
+```
+address_index | user
+--------------|-----------
+            0 | Alice
+            1 | Bob
+            2 | Alice
+            3 | Charlie
+            4 | Charlie
+            5 | Dave
+            6 | Eve
+            7 | Alice
+```
+
+Notice that each index belongs to only one user. But the same user may have one or more indexes.
+
+This approach does not scale easily because the global wallet will have to keep track of more and more addresses over time. The more users, the more addresses. The more addresses, the more memory is required.
+
+Another challenge is to scan the transactions of the global wallet. The most common stop criteria for the scanner is the GAP Limit, which stops the scanner when `N` addresses in a row are empty. An address is empty when it has no transactions.
```

Comment:
Given that it's a generic doc for use cases, I think we can explain when the gap limit problem can happen. I though about starting this paragraph like this:

Depending on the moment that the use case associates a new address to a user, we may face another challenge to know when to stop scanning the wallet address. E.g. an exchange associates the address to a user as soon as the user requests it but he might never send a transaction to it, on the other hand, some use cases can wait until the moment the user needs to send its first transaction before associating an address to him. In this scenario, no addresses would be unused.